### ISSUE TITLE
alternative loss is numerically not stable

### ISSUE THREAD

**Issue by HotaekHan** [kuangliu/pytorch-retinanet#17]
loss = -w*pt.log() / 2

because of this line, that loss function is numerically not stable.
for instance, pt.log() would be -inf when pt value is going to zero.
pt value is sigmoid encoding value. so it can be zero.

**Comment by HotaekHan** [comment]
and why don't you use cross entropy loss instaed of binary cross entropy loss function.
is there any reason to use bce loss function???

**Comment by vignesh-irtt** [comment]
Have u solved it

**Comment by HotaekHan** [comment]
oh, no. I didn't anything to try :(
but I think it can be solved with log-sum-exp trick.
pytorch's bce function used log-sum-exp trick. so I just used it.
